addons/html_editor/static/src/fields/html_field.js

```diff
@@ -351,3 +353,32 @@ export const htmlField = {
 };
 
 registry.category("fields").add("html", htmlField, { force: true });
+
+export function getHtmlFieldMetadata(content, metadataAttribute) {
+    const regex = new RegExp(`${metadataAttribute}\\s*=\\s*"([^"]+)"`);
+    return content.match(regex)?.[1];
+}
+
+export function setHtmlFieldMetadata(content, metadata) {
+    return Object.entries(metadata).reduce((newContent, [attribute, value]) => {
+        if (!value) {
+            return newContent;
+        }
+        const htmlContent = newContent.toString() || "<div></div>";
+        const parser = new DOMParser();
+        const doc = parser.parseFromString(htmlContent, "text/html");
```

Comment:
Note that this is particularly bad because you're reparsing the whole document for each attribute... ^^'